lib/MusicBrainz/Server/EditSearch/Predicate/EditNoteAuthor.pm

```diff
@@ -1,47 +1,24 @@
 package MusicBrainz::Server::EditSearch::Predicate::EditNoteAuthor;
-
 use Moose;
-use MusicBrainz::Server::Validation qw( is_database_row_id );
-
-with 'MusicBrainz::Server::EditSearch::Predicate';
-
-sub operator_cardinality_map {
-    return (
-        '=' => 1,
-        '!=' => 1,
-    );
-};
-
-sub valid {
-    my ($self) = @_;
-
-    my @args = $self->arguments;
-    return scalar(@args) == 1 && is_database_row_id($args[0]);
-}
+use namespace::autoclean;
 
-sub combine_with_query {
-    my ($self, $query) = @_;
-
-    my $sql = 'EXISTS (
+with 'MusicBrainz::Server::EditSearch::Predicate::Role::User' => {
+    template_clause => 'EXISTS (
         SELECT TRUE FROM edit_note
-         WHERE edit_note.editor = ?
+         WHERE ROLE_CLAUSE(edit_note.editor)
```

Comment:
I tried to create a subroutine at first, finally found this approach to be cleaner, but it definitely **has to** be comprehensible. Maybe a bit of renaming would make things more clear:

- `template_clause` could be renamed `context_clause`,
- `ROLE_CLAUSE` could be renamed `ROLE_OPERATOR` or `ROLE_TEST`

Or any other more convincing names… 🙄